### PR TITLE
Use proto subscript and assignment overload operators buildins

### LINKED ISSUE #62: lambda_tests3a fails on VS2015/2017
http://www.boost.org/development/tests/develop/developer/output/teeks99-09-p-win2012R2-64on64-boost-bin-v2-libs-phoenix-test-lambda_tests3a-test-msvc-14-1-dbg-adrs-mdl-64-thrdp-wn32-thrd-mlt.html
http://www.boost.org/development/tests/develop/developer/output/teeks99-09-f-win2012R2-64on64-boost-bin-v2-libs-phoenix-test-lambda_tests3a-test-msvc-14-0-dbg-adrs-mdl-64-thrdp-wn32-thrd-mlt.html

This does not fix anything:
```cpp
# define BOOST_NO_CXX11_VARIADIC_TEMPLATES
# define BOOST_NO_CXX11_RVALUE_REFERENCES
# define BOOST_PHOENIX_NO_VARIADIC_ACTOR
# define BOOST_PHOENIX_NO_VARIADIC_CALL
# define BOOST_PHOENIX_NO_VARIADIC_FUNCTION_EQUAL
# define BOOST_PHOENIX_NO_VARIADIC_FUNCTION_EVAL
# define BOOST_PHOENIX_NO_VARIADIC_PHX2_RESULT
# define BOOST_PHOENIX_NO_VARIADIC_EXPRESSION
# define BOOST_PHOENIX_NO_VARIADIC_BIND
```

It look to me like a proto bug.

### PR DESCRIPTION
**Pull request by Kojoley**
Fixes #62.

### CODE REVIEW

**Comment by djowel** [comment]
Oh no... IIRC there's a reason why we have those instead of the builtins. I have to jog my memory though. @sithhell , do you remember?

**Comment by Kojoley** [comment]
I was also feeling it is here for a reason, but failed to find one, then I gave it a try and it works, also fixing an issue...

**Comment by djowel** [comment]
> I was also feeling it is here for a reason, but failed to find one, then I gave it a try and it works, also fixing an issue...

I may be wrong. It's been a while since I looked at the code.

File: include/boost/phoenix/core/actor.hpp [hunk 1]
```diff
@@ -203,66 +203,8 @@ namespace boost { namespace phoenix
             expr_type;
 
         BOOST_PROTO_BASIC_EXTENDS(expr_type, actor<expr_type>, phoenix_domain)
-
-        // providing operator= to be assignable
-        actor& operator=(actor const& other)
-        {
-            detail::assign()(*this, other);
```
[the remark this answers is not in the document]
Comment (Kojoley) [reply include/boost/phoenix/core/actor.hpp]: Yeah, it can. Copy ctor was the last thing I have removed and missed that fact.

**Comment by Kojoley** [comment]
With this patch `actor` definition is closer to what one can find in [proto's lambda example](http://www.boost.org/doc/libs/master/doc/html/proto/users_guide.html#boost_proto.users_guide.examples.lambda).

**Comment by djowel** [comment]
I'm good with it as long as all the tests pass. The concern might be no longer relevant.

**Comment by Flast** [comment]
OK

**Pull request merged**